lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,44 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = formatted_cause(exception)
+
+          lines += potential_lines unless potential_lines.empty?
+
+          lines
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            exceptions = []
+
+            while exception.respond_to?(:cause) && exception.cause
```

Comment:
Actually, I think I'd favor not using test doubles for this case.  Exceptions aren't hard or expensive to create, and we're not designing the exception API, so I don't see much benefit to using test doubles for the exceptions here.